src/jit/morph.cpp

```diff
@@ -15130,11 +15099,8 @@ void Compiler::fgCheckQmarkAllowedForm(GenTree* tree)
 #else // LEGACY_BACKEND
     GenTreePtr colon = tree->gtOp.gtOp2;
 
-    assert(colon->gtOp.gtOp1->gtOper == GT_CNS_INT);
-    assert(colon->gtOp.gtOp1->AsIntCon()->IconValue() == 0);
-
-    assert(colon->gtOp.gtOp2->gtOper == GT_CNS_INT);
-    assert(colon->gtOp.gtOp2->AsIntCon()->IconValue() == 1);
+    assert(colon->gtOp.gtOp1->IsZero());
+    assert(colon->gtOp.gtOp2->IsIOne());
```

Comment:
Another instance of liberal check and making an assumption that certain kind/type of IR nodes never gets created.  What if this assumption gets violated and do we know that some place in JIT will catch it rather than leading to SBCG?